pkg/skaffold/verify/docker/deploy.go

```diff
@@ -0,0 +1,395 @@
+	return s.Wait()
+}
+
+// deploy creates a container in the local docker daemon from a build artifact's image.
+func (d *Deployer) deploy(ctx context.Context, out io.Writer, artifact graph.Artifact, tc latest.VerifyTestCase) error {
+	out, ctx = output.WithEventContext(ctx, out, constants.Verify, tc.Name)
+
+	if container, found := d.tracker.ContainerForImage(artifact.ImageName); found {
+		olog.Entry(ctx).Debugf("removing old container %s for image %s", container.ID, artifact.ImageName)
+		if err := d.client.Delete(ctx, out, container.ID); err != nil {
+			return fmt.Errorf("failed to remove old container %s for image %s: %w", container.ID, artifact.ImageName, err)
+		}
+		d.portManager.RelinquishPorts(container.Name)
+	}
+	containerCfg, err := d.containerConfigFromImage(ctx, artifact.Tag)
+	if err != nil {
+		return err
+	}
+
+	if len(tc.Container.Command) != 0 {
+		containerCfg.Entrypoint = tc.Container.Command
+	}
+	if len(tc.Container.Args) != 0 {
+		containerCfg.Cmd = tc.Container.Args
+	}
+	containerName := d.getContainerName(ctx, artifact.ImageName)
+	opts := dockerutil.ContainerCreateOpts{
+		Name:            containerName,
+		Network:         d.network,
+		ContainerConfig: containerCfg,
+		VerifyTestName:  tc.Name,
+	}
+
+	var debugBindings nat.PortMap
+	if d.debugger != nil {
+		debugBindings, err = d.setupDebugging(ctx, out, artifact, containerCfg)
+		if err != nil {
+			return errors.Wrap(err, "setting up debugger")
+		}
+
+		// mount all debug support container volumes into the application container
+		var mounts []mount.Mount
+		for _, m := range d.debugger.SupportMounts() {
+			mounts = append(mounts, m)
+		}
+		opts.Mounts = mounts
+	}
+
+	bindings, err := d.portManager.AllocatePorts(artifact.ImageName, d.resources, containerCfg, debugBindings)
+	if err != nil {
+		return err
+	}
+	opts.Bindings = bindings
+	// verify waits for run to complete
+	opts.Wait = true
+	opts.ContainerConfig.Env = os.Environ()
+
+	d.TrackContainerFromBuild(artifact, tracker.Container{Name: containerName, ID: tc.Name})
+
+	eventV2.VerifyInProgress(opts.VerifyTestName)
+	_, err = d.client.Run(ctx, out, opts)
+	if err != nil {
+		eventV2.VerifyFailed(tc.Name, err)
+		return errors.Wrap(err, "creating container in local docker")
+	}
+
+	eventV2.VerifySucceeded(opts.VerifyTestName)
+	return nil
+}
+
+// setupDebugging configures the provided artifact's image for debugging (if applicable).
+// The provided container configuration receives any relevant modifications (e.g. ENTRYPOINT, CMD),
+// and any init containers for populating the shared debug volume will be created.
+// A list of port bindings for the exposed debuggers is returned to be processed alongside other port
+// forwarding resources.
+func (d *Deployer) setupDebugging(ctx context.Context, out io.Writer, artifact graph.Artifact, containerCfg *container.Config) (nat.PortMap, error) {
+	initContainers, err := d.debugger.TransformImage(ctx, artifact, containerCfg)
+	if err != nil {
```

Comment:
Does this allow users to debug tests?
I am not sure if this was listed in the design document. How would user's be able to use this in CLI or IDE?